internal/provider/vsphere/inventory.go

```diff
@@ -47,3 +63,31 @@ func (p *vsphereProvider) SetPlatformUsages(
 	usageApi.Add(usages, usage.PlatformSelectionUsage, props)
 	return nil
 }
+
+func (p *vsphereProvider) GenerateManifests(cluster *common.Cluster) (map[string][]byte, error) {
+	manifests := make(map[string][]byte)
+	filename, content, err := p.generateSchedulableMastersManifest(cluster)
+	if err != nil {
+		return nil, err
+	}
+	if filename == "" {
+		return nil, errors.New("template filename cannot be empty")
+	}
+	manifests[filename] = content
+	return manifests, nil
+}
+
+func (p *vsphereProvider) generateSchedulableMastersManifest(cluster *common.Cluster) (string, []byte, error) {
```

Comment:
This methods seem duplicated 3 times.
Can we have a way to have it in only one place?

Reply:
done